csrc/ir/builder.h

```diff
@@ -100,6 +100,10 @@ class IrBuilder {
   // Get tensor metadata
   static Val* metadataExpr(TensorView* tv);
 
+  // Get tensor base address, for gmem tensor, it is something like
+  // `T1.data`. For smem tensor, it is something like `toSmem(T1)`.
+  static Val* tensorBaseAddressExpr(TensorView* tv);
```

Comment:
Naming nit: `tensor` in the name seems a bit redundant. 

```suggestion
  static Val* baseAddressExpr(TensorView* tv);
```

Reply:
Makes sense, changed